### PR TITLE
Upload CSS Functions lesson in html_css

### PR DESCRIPTION
**Pull request by rachelfearon**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The PR title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 
#### 1.Describe the changes made and include why they are necessary or important:

CSS Functions lesson upload for new HTML & CSS curriculum content, submitted by Lunchbox#2045 and Miggels#9514.

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number (and any [relevant keywords](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)) below:

#XXXXX

### CODE REVIEW

File: html_css/css_math_functions.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+##### What is a function and how are they used in CSS?
+
+As in other programming languages, functions are reusable pieces of code which perform specific tasks. Functions are passed “arguments” between parentheses, each of which is used by the function in a specific way. Some common examples are:
+
+`color: hsl(230, 100%, 50%);`
+`background: linear-gradient(90deg, blue, red);`
+
+Here, the value of `color` is the function `rgb()`, which accepts arguments in the form of numbers. It processes those numbers to calculate the rgb color corresponding to the three values given. Similarly, the `background` property has a value of `linear-gradient(90deg, blue, red)`, which calculates the angle, starting color, and ending color of a gradient.
+
+Unlike other programming languages you’ll use in TOP, CSS does not allow us to create our own functions. Instead, the language comes bundled with a list of premade functions that will help you solve the most common styling problems. [Here is a complete list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions) of all CSS functions and how they are used.
```
Comment (codyloyd) [inline html_css/css_math_functions.md:21]: I'd stick this link in the 'assignment' section at the end so we don't distract our students here.

File: html_css/css_math_functions.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+Setting main to equal the outcome of: `100vh - (3rem + 40px)`.
+To put it another way:  `main = 100vh - (header + footer)`.
+`calc()` is handling the math for us even though we are mixing vh, rem and px units.
+Combined with CSS variables, `calc()` can save us from the headache of repeating CSS rules.
+
+I encourage you to take a moment and edit the codepen. Play around with the different units and sizes of the elements by clicking here: <img src="https://imgur.com/a/9iDhtL0" alt="editCodePen">
+
+##### min()
+
+`min()` does an excellent job of helping us create responsive websites. Take a look at this example:
```
Comment (codyloyd) [inline html_css/css_math_functions.md:53]: I like this example... can we add a border to the icon to make extra clear what's happening here?

will also need `box-sizing`

so:
```
#iconHolder {
	width: min(150px, 100%);
	height: min(150px, 100%);
  box-sizing: border-box;
  border: 6px solid blue;
}
```

File: html_css/css_math_functions.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Take a look at how `calc()` is being used here:
+`:root {`
+`--header: 3rem;`
+`--footer: 40px;`
+`--main: calc(100vh - calc(var(--header) + var(--footer)));`
+`}`
+Setting main to equal the outcome of: `100vh - (3rem + 40px)`.
+To put it another way:  `main = 100vh - (header + footer)`.
+`calc()` is handling the math for us even though we are mixing vh, rem and px units.
+Combined with CSS variables, `calc()` can save us from the headache of repeating CSS rules.
```
Comment (codyloyd) [inline html_css/css_math_functions.md:47]: this is a little overly pedantic i think.... but while this is a really nice example of calc (and I can't really think of a better one at the moment.....) It's probably not the best way to actually achieve the layout you've got here.  Can we add a little disclaimer to that effect here?

**Review by codyloyd — CHANGES_REQUESTED** [review]

File: html_css/css_math_functions.md (hunk 1 above)
Comment (rachelfearon) [reply html_css/css_math_functions.md]: I added an Assignments section and added the review of this link to it.

File: html_css/css_math_functions.md (hunk 2 above)
Comment (rachelfearon) [reply html_css/css_math_functions.md:55]: I will touch base with Miggels on how to add this change.

File: html_css/css_math_functions.md (hunk 3 above)
Comment (rachelfearon) [reply html_css/css_math_functions.md:47]: Added a disclaimer to that effect.

**Pull request merged**